soccer/planning/RRTPlanner.cpp

```diff
@@ -177,13 +184,29 @@ std::unique_ptr<InterpolatedPath> RRTPlanner::generateRRTPath(
     }
     // debugLog("Generate Failed 10 times");
     return lastPath;
+    
 }
 
 vector<Point> RRTPlanner::runRRT(MotionInstant start, MotionInstant goal,
                                  const MotionConstraints& motionConstraints,
                                  const ShapeSet& obstacles, SystemState* state,
                                  unsigned shellID) {
+    
+    vector<Point> straight = runRRTHelper(start, goal, motionConstraints, obstacles, state, shellID, true);
+    if (straight.size() != 0) {
+        cout << "Straight path!" << endl;
+        return straight;
+    }
+    cout << "Regular RRT Planning" << endl;
+    return runRRTHelper(start, goal, motionConstraints, obstacles, state, shellID, false);
+}
+
+vector<Point> RRTPlanner::runRRTHelper(MotionInstant start, MotionInstant goal,
```

Comment:
There is already a Javadoc for this method in the header file. It reads:

/**
     * Helper method for runRRT(), which creates a vector of points representing 
     * the RRT path.
     */